utils.js

```diff
@@ -1842,10 +1835,97 @@ export async function parsePnpmLock(pnpmLock, parentComponent = null) {
           possibleOptionalDeps[decodeURIComponent(dpurl)] = false;
         }
       }
+
       dependenciesList.push({
         ref: decodeURIComponent(ppurl),
         dependsOn: ddeplist,
       });
+
+      // pnpm-lock.yaml contains more than root dependencies in importers
+      // TODO: verify if this is a good way to check.
+      // we do what we did above but for all the other components
+      for (const importedComponentName of Object.keys(yamlObj['importers'])) {
```

Comment:
Will the importers attribute be present in all lock files?

Reply:
The entire code is now in a try catch block, if `importers` is not present then it will throw an error.
I can reduce the scope of try.. catch block to the code I have added

Reply:
Can we check for the presence of that attribute and only iterate if it is present?

Reply:
yes I will add this